src/app/groups/recent-group-list.tsx

```diff
@@ -91,139 +118,31 @@ export function RecentGroupList() {
     )
   }
 
+  console.log("state", state);
+
+  const { starredGroupInfo, groupInfo, archivedGroupInfo } = sortGroups(state);
+
   return (
-    <ul className="grid grid-cols-1 gap-2 sm:grid-cols-3">
-      {state.groups
-        .toSorted(
-          (first, second) =>
-            (state.starredGroups.includes(second.id) ? 2 : 0) -
-            (state.starredGroups.includes(first.id) ? 1 : 0),
-        )
-        .map((group) => {
-          const details =
-            state.status === 'complete'
-              ? state.groupsDetails.find((d) => d.id === group.id)
-              : null
-          return (
-            <li key={group.id}>
```

Comment:
File was getting pretty big so I just moved this into its own component `RecentGroupListCard` so that I could re-use it for the different group lists